### DISCUSSION TITLE
3D model establishment

### DISCUSSION THREAD

**Discussion by wWL2771140158** [xcompact3d/Incompact3d#118] [Q&A] [answered]
Hello, I want to realize the linear motion of the three-dimensional sphere in the Cylinder module, but I can't describe the shape of the three-dimensional sphere. The code is as follows. Can you give me some help?
![image](https://user-images.githubusercontent.com/103632822/191158674-86b4b88d-9b2d-4b22-8626-11cab84dbaf5.png)

**Comment by mathrack** [comment, marked as answer]
Hello, please provide a complete diff between your source code and the one on github. The arguments in your subroutine `geomcomplex_cyl` do not match the ones in the code, see https://github.com/xcompact3d/Incompact3d/blob/master/src/BC-Cylinder.f90#L23.

Please update the title of the discussion, otherwise I might close it. Currently, it is very generic and does not reflect your issue.

Have you manage to run the cylinder ? What is the exact issue you are facing ? Can you compile the code ? Do you have an error when running the code ? Does the code run fine but the results are not looking good ?

**Reply by wWL2771140158** [reply]
I want to simulate the motion of a three-dimensional sphere in Cylinder, so I changed the model parameters in the z direction and changed the original cylinder equation to the sphere equation. The code can be compiled and run normally, but the results are wrong. The code modification and calculation results are as follows.
![1](https://user-images.githubusercontent.com/103632822/191180801-04c6cbc5-b51c-4029-9113-8076665b6769.jpg)
![image](https://user-images.githubusercontent.com/103632822/191181131-c7aee1f6-9ce3-42a3-89dd-d4b9c1306430.png)

**Reply by mathrack** [reply]
Dear Sir,
You have also modified the input arguments of the subroutine. Please answer all my questions.
Best regards,
Mathrack

**Reply by wWL2771140158** [reply]
Only cez=1.5 is added to the input parameter, and the total time step is changed to ilast=500， Other parameters remain unchanged.
![image](https://user-images.githubusercontent.com/103632822/191192561-aaaa7b04-1ee3-4209-a0ab-fc0117898c0e.png)
![image](https://user-images.githubusercontent.com/103632822/191192651-8cc6838f-b94e-4b28-a765-a230f189ada2.png)
![image](https://user-images.githubusercontent.com/103632822/191192744-2adaf42f-5796-4723-984b-2e3bda7837b7.png)

**Reply by pbartholomew08** [reply]
@mathrack is correct, there appears to be an additional ``nz`` added to the arguments of ``geomcolpex_cyl()``

**Reply by slaizet** [reply]
Note sure also about if (and where) zp is defined. I would suggest to do the same as in the x direction.

Best,

Sylvain

**Reply by wWL2771140158** [reply]
I wonder if Cylinder can realize the object program modeling that changes the Z direction, such as changing a cylinder into a sphere or a wave column. Their z direction is not a straight line and will change with the z coordinate

**Reply by mathrack** [reply]
Dear @wWL2771140158,
Please read carefully all the answers we have provided so far and provide a detailed answer. Otherwise I will delete the discussion.
Best regards,
Mathrack
